### PR TITLE
Fix app search sorting and filtering & migrate to webcomponent

### PR DESCRIPTION
**Pull request by cbadura**
- app search applies some lib fixes related to sorting, filtering and different pageSizes values
- migration to webcomponent is finished
- libs updated to 4.42.0

### CODE REVIEW

**Review by markuczy — APPROVED** [review]
Looks good in terms of webcomponent migration. Small change in webpack needed

File: webpack.config.js [hunk 1]
```diff
@@ -47,5 +48,17 @@ const plugins = config.plugins.filter((plugin) => !(plugin instanceof ModifyEntr
 
 module.exports = {
   ...config,
-  plugins
+  plugins,
+  output: {
+    uniqueName: 'onecx-help-ui',
```
Comment (markuczy) [inline webpack.config.js:53]: Should be `onecx-permission-ui`

**Review by markuczy — CHANGES_REQUESTED** [review]

File: src/app/onecx-permission-remote.module.ts [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
+import { initializeRouter, startsWith } from '@onecx/angular-webcomponents'
+import { AppEntrypointComponent } from './app-entrypoint.component'
+import { AngularAuthModule } from '@onecx/angular-auth'
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
+import { environment } from 'src/environments/environment'
+import { Configuration } from './shared/generated'
+
+function apiConfigProvider(configService: ConfigurationService, appStateService: AppStateService) {
+  return new PortalApiConfiguration(Configuration, environment.apiPrefix, configService, appStateService)
+}
```
Comment (markuczy) [inline src/app/onecx-permission-remote.module.ts:26]: Make sure its not duplicated in SharedModule

File: webpack.config.js (hunk 1 above)
Comment (cbadura) [reply webpack.config.js]: Done

File: src/app/onecx-permission-remote.module.ts (hunk 2 above)
Comment (cbadura) [reply src/app/onecx-permission-remote.module.ts:26]: Ok

File: src/app/shared/shared.module.ts [hunk 3]
```diff
@@ -112,12 +100,7 @@ export function apiConfigProvider(configService: ConfigurationService, appStateS
     TranslateModule
   ],
   //this is not elegant, for some reason the injection token from primeng does not work across federated module
-  providers: [
-    ConfirmationService,
-    LabelResolver,
-    { provide: DialogService, useClass: PortalDialogService },
-    { provide: Configuration, useFactory: apiConfigProvider, deps: [ConfigurationService, AppStateService] }
-  ],
+  providers: [ConfirmationService, LabelResolver, { provide: DialogService, useClass: PortalDialogService }],
```
Comment (markuczy) [inline src/app/shared/shared.module.ts:103]: Please remove provider for `DialogService`

**Review by markuczy — APPROVED** [review]

File: src/app/shared/shared.module.ts (hunk 3 above)
Comment (cbadura) [reply src/app/shared/shared.module.ts:103]: done

**Review by HenryT-CG — APPROVED** [review]

**Review by HenryT-CG — APPROVED** [review]

**Pull request merged**